src/deploy/NVA_build/noobaa_init.sh

```diff
@@ -151,16 +151,40 @@ prepare_server_pvs() {
 }
 
 prepare_mongo_pv() {
-  local dir="/mongo_data/mongo/cluster/shard1"
+  local main_dir="/mongo_data"
+  local shard_dir="/mongo_data/mongo/cluster/shard1"
+  local recursive_file="/mongo_data/recursive_file"
+  local dir_id=$(stat -c '%u' ${main_dir})
+  local current_id=$(id -u)
 
   # change ownership and permissions of mongo db path
-  ${KUBE_PV_CHOWN} mongo
+  if [ "${dir_id}" != "${current_id}" ] || [ ! -f ${recursive_file} ]; then
+    echo "uid change has been identified - will change from uid: ${dir_id} to new uid: ${current_id}"
+    time ${KUBE_PV_CHOWN} mongo ${current_id}
+    touch ${recursive_file}
+  fi
+
+  if [ ! -d ${shard_dir} ]; then
+    echo "creating shard directory: ${shard_dir}" 
+    mkdir -p ${shard_dir}
+    chgrp 0 ${shard_dir}
+    chmod g=u ${shard_dir}
+  fi
+}
 
-  mkdir -p ${dir}
-  chgrp 0 ${dir}
-  chmod g=u ${dir}
+prepare_postgres_pv() {
+  local dir="/var/lib/pgsql"
+  local dir_id=$(stat -c '%u' ${dir})    
+  local current_id=$(id -u)
+
+  # change ownership and permissions of mongo db path
+  if [ "${dir_id}" != "${current_id}" ]
+  then
+    ${KUBE_PV_CHOWN} postgres ${current_id}
```

Comment:
why didn't use `recursive_file` here too?

Reply:
just because we shouldn't have remains of none recursive runs in postgres. the recursive_file was used for cases of an old none-recursive run that updated the main directory but didn't update the rest of the files in the directory.